src/node.js

```diff
@@ -206,6 +207,13 @@ class Node {
             this.api.apiHost = addr.address
             this.api.apiPort = addr.port
 
+            if (gwmatch) {
+              this.gatewayAddr = gwmatch[2]
+              const addr = multiaddr(this.gatewayAddr).nodeAddress()
```

Comment:
How about only exposing `this.gatewayAddr = multiaddr(gwmatch[2])`. That way we have the standard multiaddr we are using everywhere and you can pull the details you need at any point.